File: tests/fixtures/edgecase_repo/lua/module2.lua
local mod1 = require("module1")

local Base2 = {}
Base2.__index = Base2

function Base2.new(name)
  local self = setmetatable({}, Base2)
  self.name = name
  return self
end

local Service2 = setmetatable({}, { __index = Base2 })

function Service2:handle(payload)
  self:validate(payload)
  return true
end

function Service2:validate(payload)
  return payload ~= nil
end

local function helper2(value)
  return value
end

return Service2

File: tests/fixtures/edgecase_repo/dart/module2.dart
import 'dart:async';

import 'module1.dart';

abstract class Handler2 {
  bool handle(String payload);
}

class Base2 {
  final String name;
  Base2(this.name);
}

class Service2 extends Base2 implements Handler2 {
  Service2(String name) : super(name);

  @override
  bool handle(String payload) {
    this.validate(payload);
    return true;
  }

  void validate(String payload) {}
}

String helper2(String value) => value;
